src/Servers/Kestrel/Core/src/Middleware/Internal/DuplexPipeStreamAdapter.cs

```diff
@@ -89,6 +109,40 @@ public override ValueTask DisposeAsync()
             return new ValueTask(_outputTask);
         }
 
+        private async Task ReadInputAsync()
+        {
+            try
+            {
+                while (true)
+                {
+                    var memory = _input.Writer.GetMemory();
+
+                    var result = await Stream.ReadAsync(memory);
+
+                    if (result == 0)
+                    {
+                        break;
+                    }
+
+                    _input.Writer.Advance(result);
+                    var flushResult = await _input.Writer.FlushAsync();
+                    if (flushResult.IsCompleted)
+                    {
+                        // flushResult should not be canceled.
+                        break;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Log?.LogCritical(0, ex, $"{GetType().Name}.{nameof(ReadInputAsync)}");
```

Comment:
Does this ever happen, should it have a `LoggerMessage.Define`? Either in [`KestrelTrace`](https://github.com/aspnet/AspNetCore/blob/master/src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.cs) or [`LoggerExtensions`](https://github.com/aspnet/AspNetCore/blob/master/src/Servers/Kestrel/Core/src/Internal/LoggerExtensions.cs)?

Reply:
We have the same pattern in WriteOutputAsync. I don't think we expect this to ever happen. https://github.com/aspnet/AspNetCore/pull/12204/files#diff-23de5f6c27fec1507959dd5e19dfe3f0R186